### PR TITLE
[ws-manager-mk2] Support deployment to ephemeral clusters

### PR DESCRIPTION
**Pull request by Furisto**
## Description
Make ws-manager-mk2 work in ephemeral clusters
- Support tls for image builder proxy
- Resolve port bind conflict (ws-daemon controller manager tried to bind to port 8080 which is also the port the grpc service binds to). We will expose metrics from the already existing ws-daemon metric endpoint instead (to be added later).
- Ensure that owner token and workspace url are set even if there is a conflict updating the workspace CR

## Related Issue(s)
n.a.

## How to test
- Create ephemeral cluster (did not see the port bind issue in preview environments). Use [this branch](https://github.com/gitpod-io/ops/pull/8240) when creating the cluster.
- ws-daemon should start up 
- You should not see tls errors during startup

## Release Notes

```release-note
None
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Review by WVerlaek — APPROVED** [review]
lgtm, one nit but feel free to unhold

File: components/ws-daemon/pkg/daemon/daemon.go [hunk 1]
```diff
@@ -147,9 +147,11 @@ func NewDaemon(config Config, reg prometheus.Registerer) (*Daemon, error) {
...
 	if config.WorkspaceController.Enabled {
 		mgr, err = ctrl.NewManager(restCfg, ctrl.Options{
-			Scheme:    scheme,
-			Port:      9443,
-			Namespace: config.Runtime.KubernetesNamespace,
+			Scheme:                 scheme,
+			Port:                   9443,
+			Namespace:              config.Runtime.KubernetesNamespace,
+			HealthProbeBindAddress: "0",
+			MetricsBindAddress:     "0",
```
[the remark this answers is not in the document]
Comment (kylos101) [reply components/ws-daemon/pkg/daemon/daemon.go:154]: @WVerlaek is this accounted for in the milestone plan?

[the remark this answers is not in the document]
Comment (kylos101) [reply components/ws-daemon/pkg/daemon/daemon.go:154]: Or perhaps resolved already?

[the remark this answers is not in the document]
Comment (Furisto) [reply components/ws-daemon/pkg/daemon/daemon.go:154]: Updated the milestone plan

**Pull request merged**